<launch>
  <arg name="input_autoware_engage" />
  <arg name="input_vehicle_engage" />
  <arg name="input_route" />
  <arg name="input_twist" />

  <arg name="output_autoware_state" />
  <arg name="output_autoware_engage" />
  <arg name="output_vehicle_engage" />

  <arg name="config_file" />

  <node pkg="autoware_state_monitor" type="autoware_state_monitor" name="autoware_state_monitor" output="screen">
    <remap from="~input/autoware_engage" to="$(arg input_autoware_engage)"/>
    <remap from="~input/vehicle_engage" to="$(arg input_vehicle_engage)"/>
    <remap from="~input/route" to="$(arg input_route)"/>
    <remap from="~input/twist" to="$(arg input_twist)"/>

    <remap from="~output/autoware_state" to="$(arg output_autoware_state)"/>
    <remap from="~output/autoware_engage" to="$(arg output_autoware_engage)"/>

    <rosparam command="load" file="$(arg config_file)" />

    <param name="update_rate" value="10.0" />
    <param name="th_arrived_distance_m" value="1.0" />
    <param name="th_stopped_time_sec" value="1.0" />
    <param name="th_stopped_velocity_mps" value="0.01" />

    <param name="disengage_on_route" value="false" />
    <param name="disengage_on_complete" value="false" />
    <param name="disengage_on_error" value="false" />
  </node>
</launch>
